lib/core/src/Cardano/Wallet/Primitive/AddressDerivation/Shelley.hs

```diff
@@ -247,184 +233,110 @@ instance SoftDerivation ShelleyKey where
 
 instance WalletKey ShelleyKey where
-    changePassphrase = changePassphraseSeq
-    publicKey = publicKeySeq
-    digest = digestSeq
-    getRawKey = getKey
-    keyTypeDescriptor _ = "seq"
-
--- | Extract the public key part of a private key.
-publicKeySeq
-    :: ShelleyKey depth XPrv
-    -> ShelleyKey depth XPub
-publicKeySeq (ShelleyKey k) =
-    ShelleyKey (toXPub k)
-
--- | Hash a public key to some other representation.
-digestSeq
-    :: HashAlgorithm a
-    => ShelleyKey depth XPub
-    -> Digest a
-digestSeq (ShelleyKey k) =
-    hash (unXPub k)
-
--- | Re-encrypt a private key using a different passphrase.
---
--- **Important**:
--- This function doesn't check that the old passphrase is correct! Caller is
--- expected to have already checked that. Using an incorrect passphrase here
--- will lead to very bad thing.
-changePassphraseSeq
-    :: Passphrase "encryption"
-    -> Passphrase "encryption"
-    -> ShelleyKey depth XPrv
-    -> ShelleyKey depth XPrv
-changePassphraseSeq (Passphrase oldPwd) (Passphrase newPwd) (ShelleyKey prv) =
-    ShelleyKey $ xPrvChangePass oldPwd newPwd prv
+    changePassphrase = \(Passphrase oldPwd) (Passphrase newPwd) (ShelleyKey prv) ->
+        ShelleyKey $ xPrvChangePass oldPwd newPwd prv
+
+    publicKey = \(ShelleyKey prv) ->
+        ShelleyKey (toXPub prv)
+
+    digest = \(ShelleyKey pub) ->
+        hash (unXPub pub)
+
+    getRawKey =
+        getKey
+
+    keyTypeDescriptor _ = "she"
 
 {-------------------------------------------------------------------------------
                          Relationship Key / Address
 -------------------------------------------------------------------------------}
 
 instance PaymentAddress 'Mainnet ShelleyKey where
-    paymentAddress (ShelleyKey k0) =
-        encodeShelleyAddress (addrSingle @'Mainnet) [xpubPublicKey k0]
-    liftPaymentAddress (KeyFingerprint k0) =
-        encodeShelleyAddress (addrSingle @'Mainnet) [k0]
+    paymentAddress paymentK = do
+        Address $ BL.toStrict $ runPut $ do
+            putWord8 (enterprise + networkId)
+            putByteString . blake2b224 $ paymentK
+      where
+        enterprise = 96
+        networkId = 1
+
+    liftPaymentAddress (KeyFingerprint fingerprint) =
+        Address $ BL.toStrict $ runPut $ do
+            putWord8 (enterprise + networkId)
+            putByteString fingerprint
+      where
+        enterprise = 96
+        networkId = 1
 
 instance PaymentAddress ('Testnet pm) ShelleyKey where
-    paymentAddress (ShelleyKey k0) =
-        encodeShelleyAddress (addrSingle @('Testnet _)) [xpubPublicKey k0]
-    liftPaymentAddress (KeyFingerprint k0) =
-        encodeShelleyAddress (addrSingle @('Testnet _)) [k0]
+    paymentAddress paymentK =
+        Address $ BL.toStrict $ runPut $ do
+            putWord8 (enterprise + networkId)
+            putByteString . blake2b224 $ paymentK
+      where
+        enterprise = 96
+        networkId = 0
```

Comment:
I see you adopted already what is going to be valid in cardano-ledger-spec in a while. `0`  network id for testnet :+1:

Reply:
According to what was last communicated indeed, although it's probably not yet merged on their side.